### ISSUE TITLE
frame_color issue

### ISSUE THREAD

**Issue by ahjstone** [dunst-project/dunst#673]
~/.config/dunst/dunstrc

[global]
separator_height = 5
separator_color = auto / frame   # same result with the 2 options

When setting a different frame_color in the [global] and [urgency_low/normal/critical] sections, and issuing a series of notifications with: 

notify-send [-u low/normal/critical] "123" "ABC"

only the first notification's frames (top, bottom, left, right) are displayed with the correct frame_color.

Once other notifications pop-up, each additional notification is missing the correct top frame_color.

Is this a bug?

- Version: 1.4.1
- Install type: package
- OS: FreeBSD 12.0-RELEASE-p11

**Comment by tsipinakis** [comment]
I'm not sure I understand what the issue is here.
Could you post your full dunstrc, what color are you seeing and what are you expecting to see?

**Comment by ahjstone** [comment]
(edited a typo in the first post).

I'm expecting each notification to be displayed with its 4 border's dedicated frame_color.

[dunstrc.txt](https://github.com/dunst-project/dunst/files/3874832/dunstrc.txt) (added the txt suffix to upload the file)

separator_color = auto
![dunst-notif-auto](https://user-images.githubusercontent.com/6587029/69341457-e3225980-0c69-11ea-837e-0c25fc373139.png)

separator_color = foreground
![dunst-notif-foreground](https://user-images.githubusercontent.com/6587029/69341458-e3baf000-0c69-11ea-8e54-ee32e9f08b7e.png)

**Comment by tsipinakis** [comment]
I just got some time to test this and now I get what you mean. `separator_color = frame` will use the color of the topmost notification.

It's not really a bug but a hastily made compromise for code clarity. I'm open to adding an option of using half the separator for the top notification frame color and the other half for the bottom one as I think you expect it do.

**Comment by ahjstone** [comment]
How about making it a real separator, like:

|............................................................|
|.................upper notification................|
|............................................................|
|--------------- bottom border  --------------|
------------------ separator ------------------
|------------------ top border  ----------------|
|.............................................................|
|...................lower notification................|
|.............................................................|

**Comment by bynect** [comment]
@fwsmit I noticed some inconsistencies with the frame coloring so I wanted to implement a more well defined behavior 

![notif-frame](https://github.com/dunst-project/dunst/assets/68197565/7fa40145-33b3-4486-9944-1205e03ee81a)_

Where the yellow are the notification frames, the purple is a (optional) "window" frame and red is the separator.

**Comment by fwsmit** [comment]
What is people don't want 3 lines in between every notification? Currently the window frame is hidden in between each notification, showing only the separator

**Comment by bynect** [comment]
> What is people don't want 3 lines in between every notification? Currently the window frame is hidden in between each notification, showing only the separator

The idea was to be able to control which part of the frame is colored (there was an issue like that). So that you can indeed have no lines or 3 or whatever you like. Something like `frames = left,right`. But I would start from having a consistent coloring and border because at the moment some lines override each other and cairo is not happy

**Comment by fwsmit** [comment]
How would you make sure to have a frame at the top and the bottom of all the notifications?

**Comment by bynect** [comment]
> How would you make sure to have a frame at the top and the bottom of all the notifications? 

I won't, the choice is up to the user.

**Comment by fwsmit** [comment]
Yeah, but how would the user configure this? If you only enable left and right border there would be no top and bottom border
